AstaZero/src/CalculateReferenceLine.cpp

```diff
@@ -0,0 +1,66 @@
+#include "CalculateReferenceLine.hpp"
+
+CalculateReferenceLine::CalculateReferenceLine(const char* openDriveFileName, double deltaS)
+	: OpenDrive(openDriveFileName) {
+	// std::ofstream file;
+	// file.open("test.csv");
+	static char strbuf[1024];
+	double x, y, hdg, laneoff;
+	XYZ tmpXYZ;
+
+
+	for (auto r : road_) {
+		// file << "lane, " << r->GetId() << ", " << 0 << ", " << 0 << ", driving" << std::endl;
+		RoadLineXYZHdg roadRefLine;
+		for (auto g : r->getGeometryVector()) {
+			auto lin = linspace(g->GetLength(), deltaS);
+			for (auto ds : lin) {
+				g->EvaluateDS(ds, &x, &y, &hdg);
+				laneoff = r->GetLaneOffset(ds + g->GetS());
+				auto reflineOffsetXY = globalReflineOffset(laneoff, hdg);
+
+				x += reflineOffsetXY[0];
+				y += reflineOffsetXY[1];
+
+				// snprintf(strbuf, sizeof(strbuf), "%lf, %lf, %lf, %lf\n", x, y, 0.0, hdg);
+				// file << strbuf;
+				tmpXYZ.x = x;
+				tmpXYZ.y = y;
+				tmpXYZ.z = 0.0;
+				roadRefLine.point.push_back(tmpXYZ);
+				roadRefLine.hdg.push_back(hdg);
+			}
+		}
+		openDriveReferenceLines.insert(std::pair<int, RoadLineXYZHdg>(r->GetId(), roadRefLine));
+	}
+}
+CalculateReferenceLine::~CalculateReferenceLine() {}
+
+std::vector<double> CalculateReferenceLine::linspace(double length, double deltaS) {
+	size_t nrOfPoints = size_t(length / deltaS);
```

Comment:
nrOfPoints maybe should be nrOfSegments or something since you add more entries to the vector than this number